libusb/libusb.h

```diff
@@ -1647,6 +1647,27 @@ enum libusb_option {
 	LIBUSB_OPTION_MAX = 4
 };
 
+/** \ingroup libusb_desc
+ * The device string type.
+ */
+enum libusb_device_string_type {
+	LIBUSB_DEVICE_STRING_MANUFACTURER,
+	LIBUSB_DEVICE_STRING_PRODUCT,
+	LIBUSB_DEVICE_STRING_SERIAL_NUMBER,
+	LIBUSB_DEVICE_STRING_COUNT // The total number of string types.
+};
+
+/** \ingroup libusb_desc
+ * The maximum length for a device string descriptor in UTF-8.
+ * 
+ * 255 max descriptor length with 2 byte header 
+ *  => 253 bytes UTF-16LE
+ *  => 126 codepoints, but last codepoint is 0
+ *  => 125 * 3 + 2
+ *  => 376 bytes
+ */
+#define LIBUSB_DEVICE_STRING_UTF8_BYTES_MAX  (384U)
```

Comment:
1) Do you think it matters having `_UTF8_` as part of the name?
Yes it is important to document it, but it is not like we're goint to have different overloads for `libusb_get_device_string` for different encodings.

I'd leave it to `LIBUSB_DEVICE_STRING_BYTES_MAX`.

2) 376 vs 384U?

Reply:
1. I really don't care either way.  Happy to remove UTF8.

2. I figured it wasn't worth saving 8 bytes in case my math turns out wrong ;)  I also learned this morning that you are not supposed to NULL terminate USB string descriptors.  From USB 2.0 spec 9.6.7:
_The UNICODE string descriptor (shown in Table 9-16) is not NULL-terminated. The string length is
computed by subtracting two from the value of the first byte of the descriptor._
I have no idea how all OS's handle a string descriptor of 255 bytes = 126.5 codepoints.  If they interpret it as 127, then we need 382 UTF8 bytes with the null terminator.  384 is definitely safe.  382 should be safe.  I am no longer sure about smaller values.  I propose staying with 384 and updating the comment as to why.